### ISSUE TITLE
Detect non-testing specs

### ISSUE THREAD

**Issue by ciaranmcnulty** [phpspec/phpspec#627]
A test needs to do at least one of the following:

1. Check a value (via a matcher)
2. Check an exception (via shouldThrow)
3. Check communication (via a Mock or Spy)

It is an open question for me whether the same applies to a spec.

It would be useful to have an (optional?) mode that highlights specs where none of those three things happen.

**Comment by GrahamCampbell** [comment]
Similar to phpunit's checks then. :)

**Comment by ciaranmcnulty** [comment]
Yes, I was looking at PHPUnit's Strict Mode and the idea appeals.

However I know that part of @everzet and others' BDD practice is to write transitional non-testing specs occasionally, that then later become tests.

**Comment by stof** [comment]
Case 1 and 2 are actually the same (``shouldThrow`` is a matcher)

**Comment by ciaranmcnulty** [comment]
@stof I'd say the case is different, the way we check it (matcher) is the same..

**Comment by angelsk** [comment]
It would be nice to have a warning if you set up a test, and then forgot to actually test anything in it :)

**Comment by avant1** [comment]
I think that would be useful feature for newcomers.
When I started using *phpsec* I've wrote non-testing tests sometimes. Well, sometimes I do it even now :smile: 

It could be implemented like not fatal warning with option to suppress them.
